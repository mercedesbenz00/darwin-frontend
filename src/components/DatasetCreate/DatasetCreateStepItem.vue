<template>
  <router-link
    v-if="step.to"
    class="dataset-create__step-item"
    :class="{ 'dataset-create__step-item--v2': isV2 }"
    active-class="dataset-create__step-item--active"
    :to="step.to"
  >
    <span class="dataset-create__step-item-index">{{ index + 1 }}</span>
    <span class="dataset-create__step-item-description">{{ step.name }}</span>
  </router-link>
  <div
    v-else
    :key="index"
    class="dataset-create__step-item"
  >
    <span class="dataset-create__step-item-index">{{ index + 1 }}</span>
    <span class="dataset-create__step-item-description">{{ step.name }}</span>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

export type Step = {
  name: string,
  to?: string
}

@Component({ name: 'dataset-create-step-item' })
export default class DatasetCreateStepItem extends Vue {
  @Prop({ required: true })
  index!: number

  @Prop({ required: false, type: Boolean, default: false })
  isV2?: boolean

  @Prop({ required: true })
  step!: Step
}
</script>

<style lang="scss" scoped>

.dataset-create__step-item {
  @include row--center;
  position: relative;
  cursor: pointer;

  .dataset-create__step-item-index {
    @include typography(lg-1, inter, bold);
  }

  &--v2.dataset-create__step-item--active {
    .dataset-create__step-item-index {
      background: $colorInteractivePrimaryDefault;
      border-color: $colorInteractivePrimaryDefault;
    }
  }
}

.dataset-create__step-item-index {
  @include row--center;
  width: 30px;
  height: 30px;
  border-radius: 15px;
  @include typography(lg-1, headlines, bold);
  color: $colorSecondaryLight;
  border: 2px solid $colorSecondaryLight;
  margin: 0 10px 0 0;
}

.dataset-create__step-item--active .dataset-create__step-item-index {
  background: $colorPrimaryLight;
  border: 2px solid $colorPrimaryLight;
  color: $colorWhite;
}

.dataset-create__step-item-description {
  @include typography(lg-1, headlines, bold);
  color: $colorSecondaryLight;
}

.dataset-create__step-item--active .dataset-create__step-item-description {
  color: $colorSecondaryDark1;
}
</style>
