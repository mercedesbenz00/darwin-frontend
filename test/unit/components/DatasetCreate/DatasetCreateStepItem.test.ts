import { createLocalVue, shallowMount, Stubs } from '@vue/test-utils'

import DatasetCreateStepItem from '@/components/DatasetCreate/DatasetCreateStepItem.vue'

const localVue = createLocalVue()

const stubs: Stubs = ['router-link']

let propsData: {
  index: number
  step: { name: string, to?: string }
}

it('matches the snapshot when step has to', () => {
  propsData = {
    index: 1,
    step: { name: 'Step 1', to: '/create/1/annotator' }
  }
  const wrapper = shallowMount(DatasetCreateStepItem, { localVue, propsData, stubs })
  expect(wrapper).toMatchSnapshot()
})

it("matches the snapshot when step doesn't have to", () => {
  propsData = {
    index: 1,
    step: { name: 'Step 2' }
  }
  const wrapper = shallowMount(DatasetCreateStepItem, { localVue, propsData, stubs })
  expect(wrapper).toMatchSnapshot()
})
