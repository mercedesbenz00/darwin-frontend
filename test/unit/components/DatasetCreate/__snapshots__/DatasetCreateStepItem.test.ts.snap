// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot when step doesn't have to 1`] = `<div class="dataset-create__step-item"><span class="dataset-create__step-item-index">2</span> <span class="dataset-create__step-item-description">Step 2</span></div>`;

exports[`matches the snapshot when step has to 1`] = `<router-link-stub active-class="dataset-create__step-item--active" to="/create/1/annotator" class="dataset-create__step-item"><span class="dataset-create__step-item-index">2</span> <span class="dataset-create__step-item-description">Step 1</span></router-link-stub>`;
